src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double MULTIPLIER_FOR_ARRAY_SIZE = 1.5;
+    private int arrayCapacity = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[arrayCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= arrayCapacity) {
+            newArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("incorrect index");
+        } else if (index < size) {
+            shiftRightAndAddElement(value, index);
+            return;
+        }
+        array[size] = value;
```

Comment:
you have all logic to add by index in your method. You don't need additional checks 
```suggestion
        } 
        shiftRightAndAddElement(value, index);
```